### ISSUE TITLE
value-keyword-case for currentColor in non SVG CSS is breaking code

### ISSUE THREAD

**Issue by bitstarr** [stylelint/stylelint#5863]
### What steps are needed to reproduce the bug?

I have a CSS declaration like this:
```css
.footer a {
  color: currentColor;
}
```
which gets reported as value-keyword-case error.

### What Stylelint configuration is needed to reproduce the bug?

```json
{
    "rules": {
        "value-keyword-case": [ "lower", {
            "ignoreProperties": [ "@font-stack", "font-family", "font" ]
        } ]
    }
}

```

### How did you run Stylelint?

CLI via ```npx stylelint assets/css/```

### Which version of Stylelint are you using?

14.3.0

### What did you expect to happen?

no warning or error because of the syntax correct value ``currentColor``

### What actually happened?

```
assets/css/…
 4:19  ✖  Expected "currentColor" to be "currentcolor"  value-keyword-case
```

When set to non camelCase, browser will not respect this value, which breaks the styling.

### Proposal to fix the bug

The root of this issue may lay in #5849. Please take a closer look.

**Comment by bitstarr** [comment]
after i read deeper in the linked PR and issues this seems to be wanted behaviour, but ```currentcolor``` seems not supported in browsers right now.
`camelCaseSvgKeyword` must be set to `true` (by default) for now, or am I wrong?

**Comment by jeddy3** [comment]
@bitstarr Thanks for the report and for using the template.

> browser will not respect this value

I was under the impression that keywords as case insensitive in CSS. Latest Chrome, Firefox and Safari seem to respect the keyword in any case: `currentColor`, `CURRENTCOLOR`, and `currentColor` etc.

Which browser and version are you seeing the issue in?

**Comment by ahallicks** [comment]
Although it does work using `currentcolor` I would argue that [the spec](https://www.w3.org/TR/css-color-3/#currentcolor) lists it as `currentColor` so should this also be respected (and ignored or tweaked in the Stylelint check)?

**Comment by jeddy3** [comment]
The [original issue](https://github.com/stylelint/stylelint/issues/4884) was opened to align it with [the level 4 spec](https://www.w3.org/TR/css-color-4/#currentcolor-color):

> The keyword `currentcolor` represents value of the color property on the same element. Unlike `<named-color>`s, it is not restricted to sRGB; the value can be any `<color>`. Its used values is determined by resolving color values.

> Note: Multi-word keywords in CSS usually separate their component words with hyphens. `currentcolor` doesn’t, because (deep breath) it was originally introduced in SVG as a property value, "current-color" with the usual CSS spelling. It (along with all other properties and their values) then became presentation attributes and attribute values, as well as properties, to make generation with XSLT easier. Then all of the presentation attributes were changed from hyphenated to camelCase, because the DOM had an issue with hyphen meaning "minus". But then, they didn’t follow CSS conventions anymore so all the properties and property values that were already part of CSS were changed back to hyphenated! `currentcolor` was not a part of CSS at that time, so remained camelCased. Only later did CSS pick it up, at which point the capitalization stopped mattering, as CSS keywords are ASCII case-insensitive.

I believe the case has been changed to lower to match other colour keywords like `lightseagreen`.

The built-in rules are usually strict by default and made more permissive with options. In this case, we default to `"lowercase"` and have added a `camelCaseSvgKeyword` option, which can be used if you want level 3 case for SVG keywords.

However, if lowercase for SVG keywords is problematic in some browsers then we may need to revisit this. It'd be good to know which browsers and what versions are affected.

**Comment by bitstarr** [comment]
Seem like my Firefox already updated today so i cannot say which version it was that did caused the trouble. Tested right now in different browsers and devices, but it seems fine. I don't know what was wrong yesterday :(

> However, if lowercase for SVG keywords is problematic in some browsers then we may need to revisit this. It'd be good to know which browsers and what versions are affected.

Maybe there are still browsers in use, that can cause trouble, but all I have at hand are interpreting both spelling fine right now.

Sorry for bothering you all.

**Comment by jeddy3** [comment]
No problem at all.

I suggest we leave this issue open for a few days so that it's easy to find should anyone else encounter a problem in the browser they use. If nothing new is reported by then, I think we can close the issue.

**Comment by portfolioris** [comment]
I've found the following:

- Firefox's devtools autocomplete to `currentColor`, but accepts both,
- Edge's devtools autocomplete to both `currentColor` and `currentcolor`,
- Webstorm autocompletes to `currentColor`,
- MDN mentions both ways on different pages,
- The spec says `currentcolor`

I think the spec is the most accurate, and browsers and IDEs need to catch up.

**Comment by semihraifgurel** [comment]
```
rules: {
    "value-keyword-case": [
        "lower",
        {
            "camelCaseSvgKeywords": true
        }
    ]
}
```

**Comment by kawaguchi1102** [comment]
@portfolioris thank you!

In a postscript:

- Chrome's devtools autocomplete to `currentColor`, but accepts both.
- Safari's devtools autocomplete to `currentcolor`, but accepts both.

**Comment by xavierfoucrier** [comment]
I agree with @jeddy3, it's not really a problem as the specification will evolve in CSS Level 4.
Right now, we also got the `camelCaseSvgKeywords` rule to prevent this.

> Note that I have tested on IE11 just to see, and it works even when  using the `currentcolor` lowercase value.

**Comment by FRSgit** [comment]
Hey! Wanted to put my 2 cents in here:
Should stylelint really try to be aligned with [CSS color level 4 spec](https://www.w3.org/TR/css-color-4/) which is a working draft (didn't get official W3C review)? Citing the definition of a Working Draft:
>These working drafts have not received formal review and are not endorsed W3C.
>
>These draft notes MUST NOT be cited as W3C standards and MAY never become W3C standards.
>
>Software MAY implement these specifications at their own risk but implementation feedback is be encouraged.
>
>There are no patent protection covering the implementations of a W3C Working Draft.

IMO it would be understandable to follow [CSS level 3 standard](https://www.w3.org/TR/css-color-3/#currentcolor) which already has a [recommendation status](https://www.w3.org/standards/types#REC). Especially knowing that browsers and IDEs encourage usage of `currentColor` (with autocompletes).

Is there any part of the `stylelint` documentation saying that `stylelint` is meant to encourage using of the features specified in W3C working drafts or recommendations? If not, maybe that should be done at some point to make it easier to decide on the issues like this one.

**Comment by jeddy3** [comment]
@FRSgit Can you raise your issue in the [standard config tracker](https://github.com/stylelint/stylelint-config-standard/issues), please? It's a good point and we can improve the documentation of that package.
